mmedit/models/base_archs/wrapper.py

```diff
@@ -0,0 +1,135 @@
+    >>> config = dict(
+    >>>     type='ControlNetModel',  # has been registered in `MODELS`
+    >>>     from_config='lllyasviel/sd-controlnet-canny',
+    >>>     cache_dir='~/.cache/OpenMMLab')
+    >>> controlnet = MODELS.build(config)
+
+    >>> 3. Initialize model with own defined arguments
+    >>> config = dict(
+    >>>     type='ControlNetModel',  # has been registered in `MODELS`
+    >>>     in_channels=3,
+    >>>     down_block_types=['DownBlock2D'],
+    >>>     block_out_channels=(20, ),
+    >>>     conditioning_embedding_out_channels=(16, ))
+    >>> controlnet = MODELS.build(config)
+
+    Args:
+        from_pretrained (Union[str, os.PathLike], optional): The *model id*
+            of a pretrained model or a path to a *directory* containing
+            model weights and config. Please refers to
+            `diffusers.model.modeling_utils.ModelMixin.from_pretrained`
+            for more detail. Defaults to None.
+        from_config (Union[str, os.PathLike], optional): The *model id*
+            of a pretrained model or a path to a *directory* containing
+            model weights and config. Please refers to
+            `diffusers.configuration_utils.ConfigMixin.load_config`
+            for more detail. Defaults to None.
+        init_cfg (dict or List[dict], optional): Initialization config dict.
+            Noted that, in `DiffuserWrapper`, if you want to load pretrained
+            weight from HuggingFace space, please use `from_pretrained`
+            argument instead of using `init_cfg`. Defaults to None.
+
+        *args, **kwargs: If `from_pretrained` is passed, *args and **kwargs
+            will be passed to `from_pretrained` function. If `from_config`
+            is passed, *args and **kwargs will be passed to `load_config`
+            function.  Otherwise, *args and **kwargs will be used to
+            initialize the model by `self._module_cls(*args, **kwargs)`.
+    """
+
+    def __init__(self,
+                 from_pretrained: Optional[Union[str, os.PathLike]] = None,
+                 from_config: Optional[Union[str, os.PathLike]] = None,
+                 init_cfg: Union[dict, List[dict], None] = None,
+                 *args,
+                 **kwargs):
+        super().__init__(init_cfg)
+
+        # import ipdb
+        # ipdb.set_trace()
+        module_cls = self._module_cls
+        assert not (from_pretrained and from_config), (
+            '\'from_pretrained\' and \'from_config\' should not be passed '
+            'at the same time.')
+
+        self._from_pretrained = from_pretrained
+        self._from_config = from_config
+
+        if from_pretrained is not None:
+            self.model = module_cls.from_pretrained(from_pretrained, *args,
+                                                    **kwargs)
+            self._is_init = True
+        elif from_config is not None:
+            self.model = module_cls.from_config(from_config, *args, **kwargs)
```

Comment:
a future warning from `diffusers`
```
FutureWarning: It is deprecated to pass a pretrained model name or path to `from_config`.If you were trying to load a model, please use <class 'diffusers.models.controlnet.ControlNetModel'>.load_config(...) followed by <class 'diffusers.models.controlnet.ControlNetModel'>.from_config(...) instead. Otherwise, please make sure to pass a configuration dictionary instead. This functionality will be removed in v1.0.0.
  deprecate("config-passed-as-path", "1.0.0", deprecation_message, standard_warn=False)
```